### DISCUSSION TITLE
How to enable Run on save?

### DISCUSSION THREAD

**Discussion by clicktodev** [rust-lang/rust-analyzer#17298] [Q&A] [answered]
Clicking the built-in `Run` button doesn't rerun when saving files.
![image](https://github.com/rust-lang/rust-analyzer/assets/34947993/4ec482fa-e307-48b9-ab4d-56f7efaf2762)

I have to use:
`cargo watch -x 'run main.rs'`

**Comment by lnicola** [comment, marked as answer]
Indeed, the Run lens isn't going to re-run the program after each save.

For programs that finish (so not Web servers) you can get an approximation of it by setting `"rust-analyzer.check.command": "run"`, but you're not going to see the terminal output. I suggest continuing to use `cargo watch`.
